core/ui_mixins/quick_panel.py

```diff
@@ -255,7 +255,7 @@ def _on_highlight(self, index):
     def _on_selection(self, index):
         if index == self.limit:
             self.skip = self.skip + self.limit
-            sublime.set_timeout(self.show, 10)
```

Comment:
You're right, frozen UI sucks, but I'm just imagining if a user may want to wait for the next result list, rather than be surprised by the callback when it comes back...

Though in most cases it will happen faster than an attention span loss, so let's use async.